src/main/java/com/mongodb/kafka/connect/sink/StartedMongoSinkTask.java

```diff
@@ -59,21 +69,40 @@ public final class StartedMongoSinkTask {
     this.mongoClient = mongoClient;
     this.errorReporter = errorReporter;
     checkedTimeseriesNamespaces = new HashSet<>();
+    statistics = MBeanServerUtils.registerMBean(new SinkTaskStatistics(), getMBeanName());
+  }
+
+  private String getMBeanName() {
+    return "com.mongodb:type=MongoDBKafkaConnector,name=SinkTask" + id;
```

Comment:
Investigating

Reply:
The [kafka documentation](https://kafka.apache.org/documentation/#common_node_monitoring) seems to have a strange MBean name and I'm wondering if there is some sort of convention based approach.

eg:
```
kafka.streams:type=stream-task-metrics,thread-id=([-.\w]+),task-id=([-.\w]+)
```

Reply:
I can't see how to access the task id - unfortunately the task context string has it in eg: `WorkerSinkTaskContext{id=MongoSinkConnector-0}` but I can't see how accessible it is.  I would reach out to the apache mailing list or confluent via Robert W and see if there is a better way to access.

Reply:
https://cwiki.apache.org/confluence/display/KAFKA/KIP-803
https://lists.apache.org/thread/o4f6qbxkl97zytg9znb2dmp867sl7hdg

Looks like its not possible but it is possible to pass an identifier via the `MongoSinkConnector#taskConfigs`  and use the index there for the task id. I think it would correlate to the actual task id.

Reply:
I'll defer this. Passing in an index via the config map doesn't seem substantively different. Both approaches would, in normal cases, increment from 0 to N and match up with Kafka's id assignment.

The id is [embedded in the thread name](https://github.com/a0x8o/kafka/blob/master/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java#L243). I'll try to pull the number out of there, even though it is internal, and fall back to the id if this fails. This seems like the best way to ensure that we match the Kafka JMX names.